src/schema_aggregator/application/enhancement/article-schema-enhancer.php

```diff
@@ -0,0 +1,224 @@
+<?php
+// phpcs:disable Yoast.NamingConventions.NamespaceName.TooLong -- Needed in the folder structure.
+namespace Yoast\WP\SEO\Schema_Aggregator\Application\Enhancement;
+
+use Exception;
+use Yoast\WP\SEO\Models\Indexable;
+use Yoast\WP\SEO\Schema_Aggregator\Domain\Enhancement\Schema_Enhancement_Interface;
+use Yoast\WP\SEO\Schema_Aggregator\Domain\Schema_Piece;
+use Yoast\WP\SEO\Schema_Aggregator\Infrastructure\Enhancement\Article_Config;
+
+/**
+ * The Article schema pieces enhancer.
+ */
+class Article_Schema_Enhancer extends Abstract_Schema_Enhancer implements Schema_Enhancement_Interface {
+
+	/**
+	 * The config.
+	 *
+	 * @var Article_Config
+	 */
+	private $config;
+
+	/**
+	 * Sets the Article_Config instance.
+	 *
+	 * @required
+	 *
+	 * @param Article_Config $config The Article_Config instance.
+	 *
+	 * @return void
+	 */
+	public function set_article_config( Article_Config $config ) {
+		$this->config = $config;
+	}
+
+	/**
+	 * Enhances specific Article schema pieces.
+	 *
+	 * @param Schema_Piece $schema_piece The schema piece to enhance.
+	 * @param Indexable    $indexable    The indexable object that is the source of the schema piece.
+	 *
+	 * @return Schema_Piece The enhanced schema piece.
+	 */
+	public function enhance( Schema_Piece $schema_piece, Indexable $indexable ): Schema_Piece {
+
+		$data = $schema_piece->get_data();
+		foreach ( $data['@graph'] as $key => $schema_data ) {
+
+			if ( isset( $schema_data['@type'] ) && ( $schema_data['@type'] === 'Article' || $schema_data['@type'] === 'NewsArticle' || $schema_data['@type'] === 'BlogPosting' ) ) {
+				$data[ $key ] = $this->enhance_schema_piece( $schema_data, $indexable );
+			}
+
+			if (
+				isset( $schema_data['@type'] ) && \is_array( $schema_data['@type'] ) && \in_array( 'Article', $schema_data['@type'], true ) ) {
```

Comment:
Since you check it twice, I'm wondering if we could move the `isset( $schema_data['@type'] )` condition to be an early return in case it is false. 
Moreover, can't the `$schema_data['@type']` array contain also `NewsArticle` or `BlogPosting`?